src/Behat/MinkExtension/Context/MinkContext.php

```diff
@@ -238,22 +239,34 @@ public function assertResponseStatusIsNot($code)
 
     /**
      * Checks, that page contains specified text.
+     * @param string  $text    the text to check
+     * @param integer $timeout in milliseconds
      *
      * @Then /^(?:|I )should see "(?P<text>(?:[^"]|\\")*)"$/
      */
-    public function assertPageContainsText($text)
+    public function assertPageContainsText($text, $timeout = 10000)
```

Comment:
In fact to solve that (and some other) problem I've developed a library http://www.qa-tools.io/ where you can set a timeout per element and then no matter when this element will be used (e.g. one or another scenario) the timeout would be checked as well.